dbms/src/Operators/LocalSortTransformOp.cpp

```diff
@@ -0,0 +1,91 @@
+// Copyright 2023 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <DataStreams/MergeSortingBlocksBlockInputStream.h>
+#include <DataStreams/SortHelper.h>
+#include <Interpreters/sortBlock.h>
+#include <Operators/LocalSortTransformOp.h>
+
+namespace DB
+{
+void LocalSortTransformOp::operatePrefix()
+{
+    header_without_constants = getHeader();
+    SortHelper::removeConstantsFromBlock(header_without_constants);
+    SortHelper::removeConstantsFromSortDescription(header, order_desc);
+    // For order by const, generate LimitOperator instead of SortOperator.
+    assert(!order_desc.empty());
+}
+
+Block LocalSortTransformOp::getMergeOutput()
+{
+    assert(merge_impl);
+    Block block = merge_impl->read();
+    if likely (block)
+        SortHelper::enrichBlockWithConstants(block, header);
+    return block;
+}
+
+OperatorStatus LocalSortTransformOp::transformImpl(Block & block)
+{
+    switch (status)
+    {
+    case LocalSortStatus::PARTIAL:
+    {
+        if unlikely (!block)
+        {
+            // convert to merge phase.
+            status = LocalSortStatus::MERGE;
+            if likely (!sorted_blocks.empty())
+            {
+                merge_impl = std::make_unique<MergeSortingBlocksBlockInputStream>(
```

Comment:
The name `MergeSortingBlocksBlockInputStream ` is confusing. 
How about `MergeSortingBlocksTransformAction`?

Reply:
The `MergeSortingBlocksBlockInputStream` is used in pull model to do merge sort.
I add a comment that `TODO refine MergeSortingBlocksBlockInputStream and use a more common class to do merge sort in both push model and pull model`.